### DISCUSSION TITLE
AppWindow - How to add a button on the Titlebar

### DISCUSSION THREAD

**Discussion by Coloris** [amwx/FluentAvalonia#515] [Q&A]
I'm wondering if there's a method to add a button in the Titlebar of AppWindow (to switch between light and dark modes), similar to what Visual Studio provides.

Checked the sample app but can't figure out for example how the AutoCompleteBox was added there.

![](https://github.com/Coloris/FluentAvaloniaNavViewTooltipIssue/blob/main/appwindow.png?raw=true)

I have this code already :

```
 TitleBar.ExtendsContentIntoTitleBar = true; 
 TitleBar.TitleBarHitTestType = TitleBarHitTestType.Complex; 
```

But how do I put a button in this area (next to the min and max button), is there any specific axaml markup that should be used or is it something done with a style ?

Thanks !

**Comment by timunie** [comment]
You just add it into your MainView I believe. There is currently no API to geht the original button positions to that's a bit tricky.

**Comment by Coloris** [comment]
It seems to be the case, and this is what has been done in the SampleApp with the AutoCompleteBox. However, I'm struggling to understand how to position a button in a similar fashion, and the documentation doesn't provide much insight into it either.

For instance, in the SampleApp, the AutoCompleteBox is centered in the middle of the Titlebar.
